source/AI.cpp

```diff
@@ -535,25 +526,44 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 	const int maxMinerCount = minables.empty() ? 0 : 9;
 	bool opportunisticEscorts = !Preferences::Has("Turrets focus fire");
 	bool fightersRetreat = Preferences::Has("Damaged fighters retreat");
+	bool shouldUpdateEscortState = !static_cast<bool>(Random::Int(20));
+	bool playerEscortStateUpdated = false;
 	for(const auto &it : ships)
 	{
 		// Skip any carried fighters or drones that are somehow in the list.
 		if(!it->GetSystem())
 			continue;
 
+		// Update fleet-wide states for NPCs, player ships, and carried ships.
+		if(shouldUpdateEscortState && !Random::Int(10) && (!it->IsYours() || !playerEscortStateUpdated))
+		{
+			if(it->IsYours())
+				playerEscortStateUpdated = true;
+			// Only update escort state if it makes sense to do so.
+			it->UpdateEscortsState();
+		}
+
+		bool isStranded = false;
+
 		if(it.get() == flagship)
 		{
 			// Player cannot do anything if the flagship is landing.
 			if(!flagship->IsLanding())
 				MovePlayer(*it, player, activeCommands);
+
+			// Your flagship may request refueling from an escort tanker
+			// carrier.  The flagship can also request help from your escorts
+			// for recharge.
+			if(!Random::Int(20) && flagship->MayRequestHelp() && (!flagship->IsDisabled() ^ flagship->IsEnergyLow()))
```

Comment:
> However, if you're disabled otherwise (such as hull damage) the classic game behavior applies and requires the player to explicitly request help.

I might be confused but (!flagship->IsDisabled() ^ flagship->IsEnergyLow()) will also trigger when you're just not disabled or in need of energy?